### PR TITLE
Showing consul errors is no longer perilous

### PR DESCRIPTION
**Pull request by rudle**

### CODE REVIEW

File: pkg/store/consul/consulutil/safe.go [hunk 1]
```diff
@@ -28,11 +23,7 @@ type KVError struct {
 
 // Error implements the error and "pkg/util".CallsiteError interfaces.
 func (err KVError) Error() string {
-	cerr := ""
-	if *showConsulErrors {
-		cerr = fmt.Sprintf(": %s", err.UnsafeError)
-	}
-	return fmt.Sprintf("%s failed for path %s%s", err.Op, err.Key, cerr)
+	return fmt.Sprintf("%s failed for path %s:%s", err.Op, err.Key, err.UnsafeError)
```
Comment (mpuncel) [inline pkg/store/consul/consulutil/safe.go]: nit: probably want a space after the `:`

File: pkg/store/consul/consulutil/safe.go [hunk 2]
```diff
@@ -7,13 +7,8 @@ import (
...
 	"github.com/hashicorp/consul/api"
-
-	"github.com/square/p2/pkg/util/param"
 )
 
-// Show detailed error messages from Consul (use only when debugging)
-var showConsulErrors = param.Bool("show_consul_errors_unsafe", false)
-
 // KVError encapsulates an error in a Store operation. Errors returned from the
 // Consul API cannot be exposed because they may contain the URL of the request,
```
Comment (petertseng) [inline pkg/store/consul/consulutil/safe.go]: I'm kind of wishing this comment would change to reflect the new reality.

... perhaps the error also no longer needs to be called `UnsafeError`...?

Comment (mpuncel) [inline pkg/store/consul/consulutil/safe.go]: also the filename probably doesn't need to be `safe.go` anymore

**Comment by bwester** [comment]
Before (effectively) permanently flipping this flag to true, has anyone verified that the token isn't leaked anymore?

**Comment by mpuncel** [comment]
I have: 


```
2017-01-26_18:00:01.60987 {"Counter":134,"PID":520194,"err":"list failed for path intent/aam12.sjc1.square: Get https://bad-consul.vip.sjc2b.square:443/v1/kv/intent/aam12.sjc1.square?recurse=\u0026stale=: dial tcp: lookup bad-consul.vip.sjc2b.square on 127.0.0.2:53: no such host","filename":"safe.go","function":"github.com/square/p2/pkg/store/consul/consulutil.SafeList.func1","level":"error","line_number":98,"msg":"there was an error reading the manifest","time":"2017-01-26T18:00:01Z"}
```

**Comment by bwester** [comment]
👍 👍 👍

File: pkg/store/consul/consulutil/safe.go (hunk 2 above)
Comment (rudle) [inline pkg/store/consul/consulutil/safe.go]: Should the funcs be called SafeGet and SafeList? Probably not.

File: pkg/store/consul/consulutil/safe.go [hunk 3]
```diff
@@ -13,17 +13,17 @@ import (
...
 type KVError struct {
-	Op          string
-	Key         string
-	UnsafeError error
-	filename    string
-	function    string
-	lineNumber  int
+	Op         string
+	Key        string
+	KVError    error
```
Comment (petertseng) [inline pkg/store/consul/consulutil/safe.go]: amazing that we can have the field name the same as the type name!

File: pkg/store/consul/consulutil/ops.go [hunk 4]
```diff
@@ -7,32 +7,23 @@ import (
...
 	"github.com/hashicorp/consul/api"
-
-	"github.com/square/p2/pkg/util/param"
 )
 
-// Show detailed error messages from Consul (use only when debugging)
-var showConsulErrors = param.Bool("show_consul_errors_unsafe", false)
-
 // KVError encapsulates an error in a Store operation. Errors returned from the
 // Consul API cannot be exposed because they may contain the URL of the request,
```
Comment (petertseng) [inline pkg/store/consul/consulutil/ops.go]: this too

File: bin/p2-inspect/main.go [hunk 5]
```diff
@@ -58,7 +58,7 @@ func main() {
 	if err != nil {
 		message := "Could not list reality kvpairs: %s"
 		if kvErr, ok := err.(consulutil.KVError); ok {
```
Comment (mpuncel) [inline bin/p2-inspect/main.go]: is this if statement necessary? what's the difference between putting a `consulutil.KVError` through `%s`

Comment (mpuncel) [inline bin/p2-inspect/main.go]: not a huge deal though

**Pull request merged**